packages/contracts/test/framework/plugin/plugin-repo.ts

```diff
@@ -61,212 +46,442 @@ describe.skip('PluginRepo', function () {
+          'PluginSetupAlreadyInPreviousRelease'
+        )
+        .withArgs(2, 1, pluginSetup_2.address);
     });
 
-    it('version is fetchable as latest', async () => {
-      assertVersion(
-        await pluginRepo.getLatestVersion(),
-        [1, 0, 0],
-        initialPluginSetup,
-        initialContent
-      );
+    it('successfully creates a version and emits correct events', async () => {
+      await expect(
+        pluginRepo.createVersion(
+          1,
+          pluginSetupMock.address,
+          BUILD_METADATA,
+          RELEASE_METADATA
+        )
+      )
+        .to.emit(pluginRepo, 'VersionCreated')
+        .withArgs(1, 1, pluginSetupMock.address, BUILD_METADATA)
+        .to.emit(pluginRepo, 'ReleaseMetadataUpdated')
+        .withArgs(1, RELEASE_METADATA);
     });
 
-    it('version is fetchable by semantic version', async () => {
-      assertVersion(
-        await pluginRepo.getVersionBySemanticVersion([1, 0, 0]),
-        [1, 0, 0],
-        initialPluginSetup,
-        initialContent
-      );
+    it('correctly increases and emits build number', async () => {
+      await expect(
+        pluginRepo.createVersion(
+          1,
+          pluginSetupMock.address,
+          BUILD_METADATA,
+          RELEASE_METADATA
+        )
+      )
+        .to.emit(pluginRepo, 'VersionCreated')
+        .withArgs(1, 1, pluginSetupMock.address, BUILD_METADATA);
+
+      expect(await pluginRepo.buildCount(1)).to.equal(1);
+
+      await expect(
+        pluginRepo.createVersion(
+          1,
+          pluginSetupMock.address,
+          BUILD_METADATA,
+          RELEASE_METADATA
+        )
+      )
+        .to.emit(pluginRepo, 'VersionCreated')
+        .withArgs(1, 2, pluginSetupMock.address, BUILD_METADATA);
+
+      expect(await pluginRepo.buildCount(1)).to.equal(2);
     });
 
-    it('version is fetchable by plugin factory address', async () => {
-      assertVersion(
-        await pluginRepo.getVersionByPluginSetup(initialPluginSetup),
-        [1, 0, 0],
-        initialPluginSetup,
-        initialContent
-      );
+    it('correctly increases and emits release number', async () => {
+      await expect(
+        pluginRepo.createVersion(
+          1,
+          pluginSetupMock.address,
+          BUILD_METADATA,
+          RELEASE_METADATA
+        )
+      )
+        .to.emit(pluginRepo, 'VersionCreated')
+        .withArgs(1, 1, pluginSetupMock.address, BUILD_METADATA);
+
+      expect(await pluginRepo.latestRelease()).to.equal(1);
+
+      // don't repeat the same plugin setup in the 2nd release
+      // otherwise it will revert.
+      const pluginSetupMock_2 = await deployMockPluginSetup();
+
+      await expect(
+        pluginRepo.createVersion(
+          2,
+          pluginSetupMock_2.address,
+          BUILD_METADATA,
+          RELEASE_METADATA
+        )
+      )
+        .to.emit(pluginRepo, 'VersionCreated')
+        .withArgs(2, 1, pluginSetupMock_2.address, BUILD_METADATA);
+
```

Comment:
We are increasing the release, but do we have a test for increasing the build to 2.2 ?

Reply:
check `correctly increases and emits build number` test.